src/visualizations/config/adapters/dhis_highcharts/series/index.js

```diff
@@ -168,10 +168,24 @@ function getDefault(series, layout, isStacked, extraOptions) {
             seriesObj.groupPadding = 0
         }
 
+        const legendSet = extraOptions?.legendSets?.find(
+            legendSet =>
+                legendSet.id === metaData.items[seriesObj.id]?.legendSet
+        )
+
         // color
-        seriesObj.color = isYearOverYear(layout.type)
-            ? indexColorPatternMap[index]
-            : idColorMap[seriesObj.id]
+        if (isYearOverYear(layout.type)) {
+            // YearOverYear: Fetch colors directly from color sets
+            seriesObj.color = indexColorPatternMap[index]
+        } else if (legendSet && legendSet.legends?.length) {
+            // Legendset: Fetch the middle color of the set
+            seriesObj.color = legendSet.legends.sort(
+                (a, b) => a.startValue - b.startValue
+            )[Math.ceil(legendSet.legends.length / 2) - 1]?.color
```

Comment:
Is there any possibility of not ending up with a color? (since the code has the `?`)

Reply:
The `else if` is checking for length so it will always find a legend. The `?` is not necessary I think.

Reply:
True. Agreed, just drop the `?`

Reply:
Good find. Done 👍